### PR TITLE
Makefile: add changelog target

### PR DESCRIPTION
**Pull request by vrothberg**
Maintaining a changelog for each new version or release of Podman helps
users to quickly skim for new changes.  Add a `make changelog` target to
facilitate creating a new log.

There are two env variables to control the base and target commit for
the new log.  The output gets prepended to the changelog.txt file, which
is a textfile in following format:
- Changelog for $(CHANGELOG_TARGET) (ISO-8601 DATE):
  * Commit subject
  * Commit subject...

Notice that the list of commit subjects excludes merge commits, and can
be manually modified after generation if needed.

`CHANGELOG_BASE=v0.3.2 CHANGELOG_TARGET=v0.3.3 make changelog` would
generate the following shortened output to the changelog.txt file:

Changelog for v0.3.3 (2018-03-17):
  * Bump to v0.3.3
  * Fix build after c/image changes
  * Update containers/image
  * Fix E2E tests
  * Address review comments
  * Fix E2E tests
  * Add restart to main podman manpage
  * Add podman restart to podman bash completions and commands

Signed-off-by: Valentin Rothberg <vrothberg@suse.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -150,6 +153,17 @@ docs: $(MANPAGES)
...
+	@echo "Creating changelog from $(CHANGELOG_BASE) to $(CHANGELOG_TARGET)"
+	$(eval TMPFILE := $(shell mktemp))
+	$(shell cat changelog.txt > $(TMPFILE))
+	$(shell echo "- Changelog for $(CHANGELOG_TARGET) ($(ISODATE)):" > changelog.txt)
+	$(shell git log --no-merges --format="  * %s" $(CHANGELOG_BASE)..$(CHANGELOG_TARGET) >> changelog.txt)
+	$(shell echo "" >> changelog.txt)
+	$(shell cat $(TMPFILE) >> changelog.txt)
+	$(shell rm $(TMPFILE))
+
+
```
Comment (mheon) [inline Makefile]: Extra whitespace here

**Comment by mheon** [comment]
I'll give a more extensive review later, but I'm definitely in favor of this - beats maintaining one by hand. Going to have to start being more aggressive about squashing commits before merges to help keep the log clean, but we should be doing that anyways

**Comment by vrothberg** [comment]
@mheon Thanks for checking! I will wait for more comments and update the PR next week.

**Comment by TomSweeneyRedHat** [comment]
Other than the nit @mheon spotted, change LGTM.  Plus it sounds like a good idea to me.

**Comment by vrothberg** [comment]
@mheon @TomSweeneyRedHat I updated the empty line and updated the PR.

**Comment by mheon** [comment]
This LGTM... If you take the WIP label off, I think we're good to merge with a few more reviews

**Comment by mheon** [comment]
bot, retest this please

**Comment by mheon** [comment]
@rh-atomic-bot r+

**Comment by mheon** [comment]
Homu is broken and this is only a makefile change. Manually merging.

**Pull request merged**